### PR TITLE
{Core} Remove useless codes in core for ai-did-you-mean-this

### PR DESCRIPTION
**Pull request by houk-ms**
**Description**  

This PR removes the useless codes in core for ai-did-you-mean-this, as the extension was already retired.

- ai-did-you-mean-this is retired: https://github.com/Azure/azure-cli-extensions/pull/2402
- azext_ai_did_you_mean_this is removed from core #15500

### CODE REVIEW

**Comment by houk-ms** [comment]
@christopher-o-toole @mirdaki for awareness

**Review by jiasli — APPROVED** [review]
I think there may be something more to remove. See #12889 for the original change.

**Comment by jiasli** [comment]
I think there may be something more to remove. See #12889 for the original change.

**Comment by mirdaki** [comment]
I have no problem with this.

**Review by yungezz — APPROVED** [review]

**Comment by houk-ms** [comment]
Some codes introduce by #12889 are not removed because they are still used in command recomendations

File: src/azure-cli-core/azure/cli/core/parser.py [hunk 1]
```diff
@@ -418,7 +392,6 @@ def _check_value(self, action, value):  # pylint: disable=too-many-statements, t
                     command_str = roughly_parse_command(cmd_list[1:])
                     ext_name = self._search_in_extension_commands(command_str)
                     if ext_name:
-                        caused_by_extension_not_installed = True
```
Comment (fengzhou-msft) [inline src/azure-cli-core/azure/cli/core/parser.py:421]: Can you override the recommended command if caused_by_extension_not_installed is true? The output is like this if users type n for dynamic extension install.

>  $ az blueprint list
> The command requires the extension blueprint. Do you want to install it now? The command will continue to run after the extension is installed. (Y/n): n
> CommandNotFoundError: The command requires the latest version of extension blueprint. To install, run 'az extension add --upgrade -n blueprint'.
> Try this: 'az account list'
> Still stuck? Run 'az --help' to view all commands or go to 'https://docs.microsoft.com/en-us/cli/azure/reference-index?view=azure-cli-latest' to learn more

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**